flake.nix

```
@@ -24,211 +25,299 @@
-        iohkNix.overlays.utils
-        iohkNix.overlays.cardano-lib
-        (final: prev: {
-          inherit flake-compat customConfig;
-          gitrev = self.rev or "dirty";
-          commonLib = lib // iohkNix.lib;
-          # cardanoLib = rec {
-          #   inherit (cardano-world.${final.system}.cardano) environments;
-          #   forEnvironments = f: lib.mapAttrs
-          #     (name: env: f (env // { inherit name; }))
-          #     environments;
-          # };
-          schema = ./schema;
-          ciJobs = self.ciJobs.${final.system};
-        })
-        (import ./nix/pkgs.nix)
-        (final: prev: {
-          fourmolu = final.haskell-nix.tool "ghc927" "fourmolu" {
-            version = "0.10.1.0";
-          };
-        })
-        self.overlay
-        # I _do not_ understand why we need it _here_, and having it haskell.nix
-        # does not work.
-        (final: prev: prev.lib.optionalAttrs prev.stdenv.hostPlatform.isMusl {
-          # this is needed because postgresql links against libicu
-          # which we build only statically (for musl), and that then
-          # needs -lstdc++ as well.
-          postgresql = prev.postgresql.overrideAttrs (old: {
-            NIX_LDFLAGS = "-lstdc++";
-          });
-        })
+      supportedSystems = [
+        "x86_64-linux"
+        "x86_64-darwin"
       ];
+    in
+      inputs.utils.lib.eachSystem supportedSystems (system:
+        let
+          nixpkgs = import inputs.nixpkgs {
+            inherit system;
+            inherit (inputs.haskellNix) config;
 
-    in eachSystem supportedSystems (system:
-      let
-        pkgs = import nixpkgs { inherit system overlays config; };
+            overlays =
+              builtins.attrValues inputs.iohkNix.overlays ++
+              [ inputs.haskellNix.overlay
 
-        inherit (pkgs.stdenv) hostPlatform;
+                (final: prev: with final; {
+                  # Required in order to build postgresql for musl
+                  postgresql = (final.postgresql_11
+                    .overrideAttrs (_: lib.optionalAttrs (stdenv.hostPlatform.isMusl) {
+                      dontDisableStatic = true;
+                      NIX_LDFLAGS = "--push-state --as-needed -lstdc++ --pop-state";
+                      # without this collate.icu.utf8, and foreign_data will fail.
+                      LC_CTYPE = "C";
+                    }))
+                    .override {
+                      enableSystemd = stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isMusl;
+                      gssSupport = stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isMusl;
+                    };
+                })
 
-        project = pkgs.cardanoDbSyncProject;
+                (_: cardanoNodeOverlay)
 
-        hlint = pkgs.callPackage pkgs.hlintCheck {
-          inherit (project.args) src;
-        };
+                (final: prev: {
+                  # The NixOS module expects these to be here
```

```suggestion
                  # The NixOS module for cardano-db-sync expects these to be here
```
I assume?

Can we add a pointer where the module is defined?